data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParserErrorListener.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.antlr.v4.runtime.ANTLRErrorListener;
+import org.antlr.v4.runtime.Parser;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.atn.ATNConfigSet;
+import org.antlr.v4.runtime.dfa.DFA;
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.List;
+
+/**
+ * @since 1.3
+ *
+ * Handles any syntaxError events that occur during parsing. All exceptions are tracked and available after parsing is complete.
+ */
+class ParserErrorListener implements ANTLRErrorListener {
+    private final List<Throwable> exceptions = new ArrayList<>();
+
+    public ParserErrorListener(final DataPrepperExpressionParser parser) {
+        parser.addErrorListener(this);
+    }
+
+    /**
+     * @since 1.3
+     *
+     * Get if any exception events were received
+     * @return if exceptions received
+     */
+    public boolean isErrorFound() {
+        return !exceptions.isEmpty();
+    }
+
+    /**
+     * @since 1.3
+     *
+     * Clears any error events received. Should be called before a new statement is parsed.
```

Comment:
Do we have a comment saying that `resetErrors` needs to be called on the parsing method?

Reply:
I added a message in the javadoc and added unit test to verify method is called.